### PR TITLE
Added tests for new FieldLenFields

### PR DESCRIPTION
**Pull request by akorb**
Added tests for last changes:
https://github.com/secdev/scapy/pull/1888

as requested by @polybassa https://github.com/secdev/scapy/pull/1888#issuecomment-471162880

Test for the **dissection** of a DTC request already existed, thus only added tests for building

### CODE REVIEW

File: scapy/contrib/automotive/obd/obd.uts [hunk 1]
```diff
@@ -784,6 +784,36 @@ assert b[0:1] == b'\x09'
...
+
+= Check building a response for Service 0A
+
+p = OBD()/OBD_S0A_DTC(dtcs=[OBD_DTC(), OBD_DTC(location='Body', code1=1, code2=7, code3=8, code4=2), OBD_DTC()])
+b = bytes(p)
+assert b[0:1] == b'\x4A'
+assert b[1:2] == b'\x03'
+assert b[2:4] == b'\x00\x00'
+assert b[4:6] == b'\x97\x82'
+assert b[6:8] == b'\x00\x00'
```
Comment (polybassa) [inline scapy/contrib/automotive/obd/obd.uts]: Thanks for this tests. But this tests still missing the default case for building. This was my initial intent to check this case.

Comment (akorb) [reply scapy/contrib/automotive/obd/obd.uts]: I think I understand. Just to be sure, you mean the following?:

```
>>> p = OBD()/OBD_S03_PR()                                                      
>>> p.show2()                                                                   
###[ On-board diagnostics ]### 
  service= EmissionRelatedDiagnosticTroubleCodesResponse
###[ S3_ResponseDTCs ]### 
     count= 0
     \dtcs\

>>> bytes(p)                                                                    
b'C\x00'
```

Comment (polybassa) [reply scapy/contrib/automotive/obd/obd.uts]: I mean
`p.dtcs == []`

Comment (akorb) [reply scapy/contrib/automotive/obd/obd.uts]: Just commited changes. Is it fine that way?

**Comment by polybassa** [comment]
Approved from my side

**Pull request closed without merging**